README.md

## Layout


### Installation
to install layout component use this command:
```powershell
  npm i @ReactSeam/layout
```
### Usage
```javascript
    import {LayoutNCHBF} from '@ReactSeam/layout';
    [OR]
    import {LayoutHNCBF} from '@ReactSeam/layout';
```
There are two types of Layout 

![alt text](./LayoutHNCBF.JPG)
![alt text](./LayoutNCHBF.JPG)

you can choose between them

### examples
```javascript
    		<LayoutNCHBFDemo header={<h5>Header</h5>}
    		nav={
    			<ul>
    				<li>link1</li>
    				<li>link2</li>
    				<li>link3</li>
    				<li>link4</li>
    			</ul>
    		}
    		footer={<h5>Footer</h5>}
    		bodyMinWidth={50} navMinWidth={5} navWidth={15} isPercentage={true}>
    			Test
    		</LayoutNCHBFDemo>
    		}
```
```javascript
    		<LayoutNCHBF header={<h5>Header</h5>}
    		nav={
    			<ul>
    				<li>link1</li>
    				<li>link2</li>
    				<li>link3</li>
    				<li>link4</li>
    			</ul>
    		}
    		footer={<h5>Footer</h5>}
    		bodyMinWidth={50} navMinWidth={5} navWidth={15} isPercentage={true}>
    			Test
    		</LayoutNCHBF>
```
### API
### Demo

### Archive
- [version 0.1.0](./CHANGELOG.md#0.1.0)





CHANGELOG.md
### Versions

### [0.1.0](#0.1.0)
#### Breaking changes
 - change usage of component
     - use properties (header footer nav) instead of <div> with attribute id
```javascript
		import LayoutHNCBF from '@ReactSeam/layout';
						export const foo = () =>
						{
								return (
									<LayoutNCHBFDemo bodyMinWidth={50} navMinWidth={30} navWidth={50} isPercentage={true}>
											<div key="header">
												<h5>Header</h5>
											</div>
											<div key="navigation">
												<a>link1</a>
												<a>link2</a>
												<a>link3</a>
												<a>link4</a>
											</div>
											<div key="body">
												{table}
											</div>
											<div key="footer">
												<h5>Footer</h5>
											</div>
										</LayoutNCHBFDemo>
								)
			}
```
change to:

```javascript

			import LayoutHNCBF from '@ReactSeam/layout';
                export const foo = () =>
                {
                    return (
							<LayoutNCHBFDemo header={<h5>Header</h5>}
                        		nav={
                        			<ul>
                        				<li>link1</li>
                        				<li>link2</li>
                        				<li>link3</li>
                        				<li>link4</li>
                        			</ul>
                        		}
                        		footer={<h5>Footer</h5>}
                        		bodyMinWidth={50} navMinWidth={5} navWidth={15} isPercentage={true}>
                        			test
							</LayoutNCHBFDemo >
                    )
    			}
  ```
- change documentation


src/stories.js
import React from 'react';
// Import the storybook libraries
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
// Import our component from this folder
import LayoutNCHBFDemo from './LayoutNCHBFDemo';
import LayoutHNCBFDemo from './LayoutHNCBFDemo';

// Here we describe the stories we want to see of the Button. The component is
// pretty simple so we will just make two, one with text and one with emojis
// Simple call storiesOf and then chain .add() as many times as you wish
//
// .add() takes a name and then a function that should return what you want
// rendered in the rendering area
const table = 					<table border="1">
					<thead>
					<tr>
						<th>
							عنوان
						</th>
						<th>
							عنوان
						</th>
						<th>
							عنوان
						</th>
						<th>
							عنوان
						</th>
					</tr>
					</thead>
						<tbody>
							<tr>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
							</tr>
							<tr>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
							</tr>
							<tr>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
							</tr>
							<tr>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
							</tr>
							<tr>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
							</tr>
							<tr>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
							</tr>
							<tr>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
							</tr>
							<tr>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
							</tr>
							<tr>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
							</tr>
							<tr>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
							</tr>
							<tr>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
								<td>
									مقدار
								</td>
							</tr>
							<tr>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
						</tr>
						<tr>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
						</tr>
						<tr>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
						</tr>
						<tr>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
							<td>
								مقدار
							</td>
						</tr>
						</tbody>
					</table>
storiesOf('LayoutNCHBF')
	.add('base', () => (
		<LayoutNCHBFDemo header={<h5>Header</h5>}
		nav={
			<ul>
				<li>link1</li>
				<li>link2</li>
				<li>link3</li>
				<li>link4</li>
			</ul>
		}
		footer={<h5>Footer</h5>}
		bodyMinWidth={50} navMinWidth={5} navWidth={15} isPercentage={true}>
			{table}
		</LayoutNCHBFDemo>
	))
	.add('narrow', () => (
		<LayoutNCHBFDemo bodyMinWidth={85} navMinWidth={0} navWidth={5} isPercentage={true}>
			<div key="header">
				<h5>Header</h5>
			</div>
			<div key="navigation">
				<a>link1</a>
				<a>link2</a>
				<a>link3</a>
				<a>link4</a>
			</div>
			<div key="body">
				{table}
			</div>
			<div key="footer">
				<h5>Footer</h5>
			</div>
		</LayoutNCHBFDemo>
	))
	.add('expanded', () => (
		<LayoutNCHBFDemo bodyMinWidth={50} navMinWidth={30} navWidth={50} isPercentage={true}>
			<div key="header">
				<h5>Header</h5>
			</div>
			<div key="navigation">
				<a>link1</a>
				<a>link2</a>
				<a>link3</a>
				<a>link4</a>
			</div>
			<div key="body">
				{table}
			</div>
			<div key="footer">
				<h5>Footer</h5>
			</div>
		</LayoutNCHBFDemo>
	))
	storiesOf('LayoutHNCBF')
	.add('base', () => (
		<LayoutHNCBFDemo
			header={
			<div key="header">
				<h5>Header</h5>
			</div>}
			nav={
					<ul>
						<li>link1</li>
						<li>link2</li>
						<li>link3</li>
						<li>link4</li>
					</ul>
				}
			footer={
				<h5>Footer</h5>
			}
			>
			<div key="body">
				{table}
			</div>
		</LayoutHNCBFDemo>
	))

